beacon-chain/rpc/beaconv1/state.go

```diff
@@ -56,3 +100,80 @@ func (bs *Server) GetStateFork(ctx context.Context, req *ethpb.StateRequest) (*e
 func (bs *Server) GetFinalityCheckpoints(ctx context.Context, req *ethpb.StateRequest) (*ethpb.StateFinalityCheckpointResponse, error) {
 	return nil, errors.New("unimplemented")
 }
+
+func (bs *Server) headStateRoot(ctx context.Context) ([]byte, error) {
+	stateRoot, err := bs.ChainInfoFetcher.HeadRoot(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to obtain root: %v", err)
+	}
+	return stateRoot, nil
+}
+
+func (bs *Server) genesisStateRoot(ctx context.Context) ([]byte, error) {
+	stateRoot, err := bs.ChainStartFetcher.PreGenesisState().HashTreeRoot(ctx)
```

Comment:
This is a lot to do whenever we need the state root, can we retrieve this any way faster? I think we can get the genesis block no?